subt/ros/robot/launch/k2.launch

```diff
@@ -1,96 +1,92 @@
+      <param name="streams" value="front_depth:depth:160:120:80.9344635009765:59.733184814453:95.5871505737305:95.5871505737305,rear_depth:depth:160:120:79.775161743164:59.8928451538085:95.64727783203125:95.64727783203125,front_scan:scan:-135:135:0.1:10.0,rear_scan:scan:-135:135:0.1:10.0,tracking_pose:pose"/>
+    </node>
 
-	<!-- REAL SENSE /-->	
-        <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" if="$(arg record_sensors)">
-		<arg name="camera" 		value="camera"/>
-		<arg name="serial_no" value="920312070828"/>
-		<arg name="enable_fisheye"      value="false"/>
+    <!-- Remove noise -->
+    <node name="depth_denoiser_front" pkg="proxy" type="depth_filter">
+      <param name="max_support_distance" value="0.6"/>
+      <param name="min_support_fraction" value="0.75"/>
+      <param name="max_depth" value="10.0"/>
+      <remap from="depth_denoiser_front/input" to="front_depth/depth"/>
+      <remap from="depth_denoiser_front/output" to="front_depth/clean_depth"/>
+    </node>
+    <node name="depth_denoiser_rear" pkg="proxy" type="depth_filter">
+      <param name="max_support_distance" value="0.6"/>
+      <param name="min_support_fraction" value="0.75"/>
+      <param name="max_depth" value="10.0"/>
+      <remap from="depth_denoiser_rear/input" to="rear_depth/depth"/>
+      <remap from="depth_denoiser_rear/output" to="rear_depth/clean_depth"/>
+    </node>
 
-  		<arg name="json_file_path"      value="$(find robot)/params_common/d435i.json"/>
-		<arg name="depth_width"         value="848"/>
-  		<arg name="depth_height"        value="480"/>
-  		<arg name="enable_depth"        value="true"/>
+    <!-- Merge depth, camera_info and lidar streams. -->
+    <node pkg="topic_tools" type="relay" name="depth_merge_front" args="front_depth/clean_depth all_depth/depth"/>
+    <node pkg="topic_tools" type="relay" name="depth_merge_rear" args="rear_depth/clean_depth all_depth/depth"/>
+    <node pkg="topic_tools" type="relay" name="camera_info_merge_front" args="front_depth/camera_info all_depth/camera_info"/>
+    <node pkg="topic_tools" type="relay" name="camera_info_merge_rear" args="rear_depth/camera_info all_depth/camera_info"/>
+    <node pkg="topic_tools" type="relay" name="scan_merge_front" args="front_scan all_scan"/>
+    <node pkg="topic_tools" type="relay" name="scan_merge_rear" args="rear_scan all_scan"/>
+  </group>
 
-  		<arg name="enable_infra1"       value="false"/>
-  		<arg name="enable_infra2"       value="false"/>
+  <node pkg="proxy" type="traversability" name="local_map" output="screen">
+    <param name="publish_rate" value="0.2"/>
+    <param name="num_threads" value="3"/>
+    <param name="blind_columns_left" value="20"/>
+    <param name="depth_image_stride" value="1"/>
+    <param name="horizontal_pixel_offset" value="3"/>
+    <param name="vertical_pixel_offset" value="3"/>
+    <param name="synthetic_obstacle_distance" value="0.64"/>
```

Comment:
Can you calculate or measure how far from the camera is the point on the ground visible in the middle of the bottom row of the depth image? That distance plus a centimetre or two goes here.

Reply:
There could be a complication, because the Z coordinate is different. I will use some compromise.